specs/ov_management_api.md

```diff
@@ -0,0 +1,171 @@
+- `invalid_number_of_vouchers`: when the value of `X-Number-Of-Vouchers` does not match the number of parsed Ownership Vouchers. `error_details` contains the key `parsed`, with an integer containing the number of Ownership Vouhcers that were encountered.
+- `unowned_voucher`: when an Ownership Voucher was uploaded for which the current Owner is not the Owner key of the server. `error_details` contains the key `unowned`, which is a list of indexes of Ownership Vouchers that have invalid ownership.
+- `invalid_voucher_signatures`: when an Ownership Voucher was uploaded for which one of the cryptographic verifications failed. `error_details` contains the key `invalid`, which contains a list of objects with the key `index` describing the index of the failing voucher, and `description` containing a string description of what failed to verify on the voucher.
+
+
+### Example
+
+This assumes a URI base of `/management`, and an authentication method of `OAuth2 bearer token`.
+
+#### Request
+
+``` HTTP
+POST /management/v1/ownership_voucher HTTP/1.1
+Host: fdo.example.com
+User-Agent: FDO-Client/1.0
+Authorization: Bearer some-token-here
+X-Number-Of-Vouchers: 3
+Content-Type: application/cbor
+Accept: application/json
+
+<voucher-1-bytes><voucher-2-bytes><voucher-3-bytes>
+```
+
+#### Succesful response
+
+``` HTTP
+HTTP/1.1 201 Created
+Content-Type: application/json
+Server: FDO-Owner-Server/1.0
+
+[{“guid”: “ec1c2515-98f4-40f6-a880-8db987a423c1”}, {“guid”: “d2e4fc0e-fc0b-42a0-8767-6e82bb58dd86”,}, {“guid”: “ccb05292-cba6-484f-b803-55d66c50887b”}]
+```
+
+#### Failed response: incomplete_voucher
+
+``` HTTP
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Server: FDO-Owner-Server/1.0
+
+{"error_code": "incomplete_voucher", "error_details": {"parsed_correctly": 2}}
+```
+
+#### Failed response: unowned_voucher
+
+``` HTTP
+HTTP/1.1 400 Bad Request
+Content-Type: application/json
+Server: FDO-Owner-Server/1.0
+
+{"error_code": "unowned_voucher", "error_details": {"unowned": ["4fd43ba9-12ec-4f32-bda0-d5c0956a19be"]}}
+```
+
+## Ownership Voucher delete
+
+HTTP Request context: `POST $base/ownership_voucher/delete`.
+
+This endpoint can be used to request the Owner Onboarding Server to delete a set of Ownership Vouchers, and to stop taking ownerhsip of the devices.
+The request body consists of a JSON list of GUIDs for which the Ownership Vouchers should get deleted.
+
+A succesful response contains an empty body.
+
+### Error codes
+
+- `unknown_device`: at least one of the GUIDs that were submitted were unknown to this Owner Onboarding Service. `error_details` contains the key `unknown`, which contains a JSON list of GUIDs that were unknown to this server.
+
+### Example
+
+This assumes a URI base of `/management`, and an authentication method of `OAuth2 bearer token`.
```

Comment:
Should we add that the URI base includes as well the API version?

Reply:
Good idea, will do.